kgforge/core/archetypes/store.py

```diff
@@ -597,3 +600,39 @@ def replace(match: Match) -> str:
     if context.has_vocab():
         pfx = "\n".join([pfx, f"PREFIX : <{context.vocab}>"])
     return f"{pfx}\n{qr}"
+
+
+def resources_from_construct_query(data, context):
```

Comment:
fun, we have the same one! https://github.com/BlueBrain/nexus-forge/blob/865ed625b843acd110c9a84be9fbc4fc630a3cba/kgforge/core/commons/sparql_query_builder.py#L198

Reply:
ha! I guess it was a "clear need"

Reply:
for this and the next, what would be the cleanest solution? I would like to use your code as it looks more organized, and I agree it fits well in the SPARQLQueryBuilder

Reply:
Cleanest solution in what sense?

(I've put the resource building methods in SparqlQueryBuilder following @MFSY 's comment, but I am of the opinion that there should be another Class for sparql response parsing, as it has nothing to do with query building. I guess the query builder can be treated as a set of helper functions for sparql, in which case maybe renaming it would be better)

Reply:
Yes, the latter is what it's happening. I would not mind also to have just a class for parsing. 
The question is ... should I keep things as I have, and whenever your branch is implemented I make another PR to use it?

Reply:
Yes I'm not too sure about that, that's what I had meant on slack that i'd be reluctant to start an implementation for the read only store because it can benefit from work in our 2 branches

Reply:
I think it's a matter of which is merged first. But I don't think you should try to make the changes on this branch fit the ones on mine, that will happen on the merge to master